lib/gcp/src/globalAddress.ts

```diff
@@ -147,9 +149,16 @@ export function* ensureAddressDoesNotExist({
     log.info(`Global Address is ${address.status}`);
 
     try {
-      yield call(deleteAddress, {
-        name: addressName
-      });
+      // Workaround for https://github.com/opstrace/opstrace/issues/976.
+      // Avoid issuing a second delete request for this resource otherwise we'll
+      // get a 400 back.
+      if (!deleteIssued) {
+        log.info(`Deleting Global Address`);
+        yield call(deleteAddress, {
+          name: addressName
+        });
+        deleteIssued = true;
```

Comment:
ok! :)

For future: another option is to issue the DELETE API call in a loop and potentially more than once and catch-log all resulting API errors (getting said 400 response is absolutely fine -- we just crashed too much). And to then have a definite 'desired state reached' criterion for leaving said loop.

That's the paradigm I have implemented via https://github.com/opstrace/opstrace/blob/ebd5c980b51877680bf25d161087035bc45c7d03/lib/aws/src/resource.ts#L148 for most of AWS resource deletion 

https://github.com/opstrace/opstrace/blob/ebd5c980b51877680bf25d161087035bc45c7d03/lib/aws/src/resource.ts#L269